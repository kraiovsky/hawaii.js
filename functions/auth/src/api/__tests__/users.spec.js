const { createMockContext } = require('@shopify/jest-koa-mocks')
const { createUserMock } = require('../../../../users/__mocks__/v1')
const Users = require('../users')
const tesCases = require('./users.cases')

let ctx
beforeEach(async () => {
  ctx = createMockContext({
    state: {
      config: { usersApiUrl: 'http://localhost:5001' },
      serviceToken: '',
    },
    header: { 'x-request-id': '123-abc-456' },
  })
  ctx.response.get = jest.fn(header => ctx.headers[header])
  createUserMock()
})

describe('create()', () => {
  describe.each(tesCases.create.happy)('👍', (email, resStatusCode, expected) => {
    test(`should ${expected}`, async () => {
      ctx.request.body = { email }
      const { statusCode, body } = await Users.create(ctx)
      expect(body).toMatchSnapshot()
      expect(statusCode).toBe(resStatusCode)
    })
  })
})
