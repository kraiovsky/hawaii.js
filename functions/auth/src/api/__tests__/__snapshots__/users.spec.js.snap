// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create() 👍 should return existing user object and status 200 1`] = `
Object {
  "data": Object {
    "attributes": Object {
      "avatar": null,
      "email": "admin@domain.com",
      "name": "Admin",
      "role": "admin",
    },
    "id": "0ae76c84-d939-456d-bb35-f60b7659ed6b",
    "links": Object {
      "self": "/users/0ae76c84-d939-456d-bb35-f60b7659ed6b",
    },
    "type": "user",
  },
  "jsonapi": Object {
    "version": "1.0",
  },
  "links": Object {
    "index": "/users",
  },
}
`;

exports[`create() 👍 should return new user object and status 201 1`] = `
Object {
  "data": Object {
    "attributes": Object {
      "avatar": null,
      "email": "user@domain.com",
      "name": null,
      "role": "user",
    },
    "id": "e9ad2f8f-b871-4048-95ff-29577f4285f5",
    "links": Object {
      "self": "/users/e9ad2f8f-b871-4048-95ff-29577f4285f5",
    },
    "type": "user",
  },
  "jsonapi": Object {
    "version": "1.0",
  },
  "links": Object {
    "index": "/users",
  },
}
`;
